support-payment-api/src/main/scala/services/SupporterProductDataService.scala

```diff
@@ -0,0 +1,58 @@
+package services
+
+import cats.data.EitherT
+import com.gu.supporterdata.model.{ContributionAmount, SupporterRatePlanItem}
+import com.gu.supporterdata.model.Stage.{DEV, PROD}
+import com.gu.supporterdata.services.SupporterDataDynamoService
+import model.db.ContributionData
+import model.Environment
+import model.Environment.Live
+
+import java.time.LocalDate
+import scala.concurrent.{ExecutionContext, Future}
+
+class SupporterProductDataService(environment: Environment) {
+  val dynamoService = SupporterDataDynamoService(environment match {
+    case Live => PROD
+    case _ => DEV
+  })
+  def insertContributionData(
+      contributionData: ContributionData,
+  )(implicit executionContext: ExecutionContext): EitherT[Future, String, Unit] =
+    for {
+      item <- EitherT.fromEither[Future](
+        contributionData.identityId
+          .map(idAsLong =>
+            SupporterRatePlanItem(
+              subscriptionName = s"${contributionData.paymentProvider.entryName} - ${contributionData.paymentId}",
```

Comment:
is `subscriptionName` likely to be used for anything?

Reply:
It's basically just to have a human readable field so we can make sense of the data in the dynamo table